src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Instance/InstanceDiscovery.cs

```diff
@@ -110,15 +111,19 @@ public static string FormatAuthorizeEndpoint(string host, string tenant)
             return string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/oauth2/authorize", host, tenant);
         }
 
+        public static string GetTenant(Uri uri)
+        {
+            return uri.AbsolutePath.Split('/')[1];  // Will generate exception when uri contains no tenant
```

Comment:
we should have parameter checking here.
null check
split.length >=1

Reply:
I can add some `if(...) throw new SomeException(...)` at the beginning of this function, but before that I just want to make sure: Do we use such pattern simply because it is a convention?

In this particular case, the current implementation is a one-liner. So, if the input `uri` happens to be `null`, it would automatically throw a `System.NullReferenceException` and the stack trace would be at exact the same line. Also if for whatever reason (more on this later) the "split.length" is less than 2, the `...[1]` part would also automatically throw a `System.IndexOutOfRangeException` and again the stack trace would be at the same line.

Reply:
Throwing a System.NullReferenceException is not good practice, in my opinion, and usually means a programming error. 
I think it is better to recognize that the uri is not well formed and pass a meaningful message back to the user.

Reply:
This uri path validation happens way before Instance discovery is called. If we are here in the code, then the input was already deemed sane.